api/s2n.h

```diff
@@ -280,8 +280,15 @@ S2N_API
 extern void *s2n_connection_get_ctx(struct s2n_connection *conn);
 
 typedef int s2n_client_hello_fn(struct s2n_connection *conn, void *ctx);
+typedef enum { S2N_CLIENT_HELLO_CB_BLOCKING, S2N_CLIENT_HELLO_CB_NONBLOCKING } s2n_client_hello_cb_mode;
 S2N_API
 extern int s2n_config_set_client_hello_cb(struct s2n_config *config, s2n_client_hello_fn client_hello_callback, void *ctx);
+S2N_API
+extern int s2n_config_set_client_hello_cb_mode(struct s2n_config *config, s2n_client_hello_cb_mode cb_mode);
+S2N_API
+extern int s2n_client_hello_cb_done(struct s2n_connection *conn);
+S2N_API
+extern int s2n_client_hello_server_name_extension_used(struct s2n_connection *conn);
```

Comment:
purpose of this fn is for the server to be able to mark that server_name extension was used to determine properties on the connection. This allows server to determine if it should send server_name extension back to the client as part of ServerHello.

I think this flag should be ok to be specific to client_hello (as that's how we determine SNI extension).

Reply:
So what if we send the SNI extension for a different reason later? Do we add a different API that does the same thing?

Reply:
this fn in not specific to callback callback. It just expects that client_hello processing must not have completed by the time this is invoked, after which we would send the ServerHello any way. 

If later we do have some other flags that determine server hello should be sent or not, i would prefer them to be explicit as that can then used to give s2n context on why server_hello extension was disabled/enabled.

Reply:
Since this isn't specific to the callback, can be capture that in it's name? `s2n_connection_enable_server_name_extension` or `s2n_connection_server_name_extension_used` instead.

Reply:
done.